packages/stark-core/src/routing/services/routing.service.spec.ts

```diff
@@ -0,0 +1,2271 @@
+			},
+			parent: "page-03",
+			views: {
+				"initOrExit@": {
+					component: LogoutPageComponent
+				}
+			}
+		}
+	];
+
+	function TestStateTreeParams(navigationSteps: any, expectedStateTreeParams: any, previousNavigations?: number): void {
+		for (const navigationStep of navigationSteps) {
+			routingService.navigateTo(navigationStep.stateName, navigationStep.stateParams);
+			tick();
+		}
+
+		if (previousNavigations) {
+			for (let i: number = 0; i < previousNavigations; i++) {
+				routingService.navigateToPrevious();
+				tick();
+			}
+		}
+
+		const stateTreeParams: Map<string, any> = routingService.getStateTreeParams();
+
+		expect(stateTreeParams.size).toBe(expectedStateTreeParams.length);
+		let index: number = 0;
+		stateTreeParams.forEach((stateParams: any, stateName: string) => {
+			expect(stateName).toBe(expectedStateTreeParams[index].stateName);
+			expect(stateParams).toEqual(expectedStateTreeParams[index].stateParams);
+			index++;
+		});
+	}
+
+	function TestStateTreeResolves(navigationSteps: any, expectedStateTreeResolves: any): void {
+		for (const navigationStep of navigationSteps) {
+			routingService.navigateTo(navigationStep.stateName, navigationStep.stateParams);
+			tick();
+		}
+
+		const stateTreeResolves: Map<string, any> = routingService.getStateTreeResolves();
+
+		expect(stateTreeResolves.size).toBe(expectedStateTreeResolves.length);
+		let index: number = 0;
+
+		stateTreeResolves.forEach((stateResolve: any, token: string) => {
+			expect(token).toBe(expectedStateTreeResolves[index].name);
+			expect(stateResolve).toEqual(expectedStateTreeResolves[index].resolvableData);
+			index++;
+		});
+	}
+
+	function TestStateTreeData(navigationSteps: any, expectedStateTreeData: any): void {
+		for (const navigationStep of navigationSteps) {
+			routingService.navigateTo(navigationStep.stateName);
+			tick();
+		}
+
+		const stateTreeData: Map<string, any> = routingService.getStateTreeData();
+
+		expect(stateTreeData.size).toBe(expectedStateTreeData.length);
+		let index: number = 0;
+
+		stateTreeData.forEach((stateData: any, stateName: string) => {
+			expect(stateName).toBe(expectedStateTreeData[index].name);
+			expect(stateData).toEqual(expectedStateTreeData[index].data);
+			index++;
+		});
+	}
+
+	function TestTranslationKey(navigationSteps: any, expectedTranslationKeys: any): void {
+		for (const navigationStep of navigationSteps) {
+			routingService.navigateTo(navigationStep.stateName);
+			tick();
+		}
+
+		for (const expectedTranslationKey of expectedTranslationKeys) {
+			const translationKey: string = routingService.getTranslationKeyFromState(expectedTranslationKey.name);
+			expect(translationKey).toBe(expectedTranslationKey.translationKey);
+		}
+	}
+
+	const routerModule: UIRouterModule = UIRouterModule.forRoot({
+		useHash: true,
+		states: mockStates,
+		deferIntercept: true // FIXME: this option shouldn't be used but is needed for Chrome and HeadlessChrome otherwise it doesn't work. Why?
```

Comment:
Indeed, in the meantime we disable this option to prevent the UI-Router from listening to URL changes so that it doesn't trigger a new navigation to the new URL.

Strangely enough, the tests pass fine Chrome 56, Firefox and Edge, however they fail in Chrome 60+ and HeadlessChrome. 

We should investigate this later on in order to remove the "deferIntercept" option from these tests.